### PR TITLE
Add page abstaction task

### PR DESCRIPTION
**Pull request by JRojas999**

### CODE REVIEW

File: src/test/java/salesforce/assets/CreateAssetTest.java [hunk 1]
```diff
@@ -11,52 +11,75 @@
...
+                .clickRoleFirstOption()
+                .clickField("Contact")
+                .clickContactFirstOption()
+                .clickField("Product")
+                .clickProductFirstOption();
+        assetDetailPage = createAssetPage.clickSaveBtn();
+        softAssert.assertEquals(assetDetailPage.getField("Account"), "cuenta 13",
+                "Account name does not match");
+        softAssert.assertEquals(assetDetailPage.getField("Contact"), "contact 2",
+                "Account name does not match");
```
Comment (Raymundogs) [inline src/test/java/salesforce/assets/CreateAssetTest.java]: each message should be different considering it is a different field.

File: src/main/java/salesforce/ui/pages/assets/NewAssetPage.java [hunk 2]
```diff
@@ -72,89 +73,137 @@
...
+     * @param optionField is second webElement that will be clicked.
+     * @return a NewAssetPage entity.
+     */
+    public NewAssetPage clickOptionField(final String field, final String optionField) {
+        webElementAction.clickOptionField(driver.findElement(By.xpath(String.format(COMMON_LOCATOR, field))),
+                driver.findElement(By.xpath(String.format(SELECT_LOCATOR, optionField))));
+        return this;
+    }
+
+    public NewAssetPage selectField(final String field) {
```
Comment (Raymundogs) [inline src/main/java/salesforce/ui/pages/assets/NewAssetPage.java:99]: each method should have its own javadoc.

File: src/main/java/salesforce/ui/pages/assets/AssetDetailPage.java [hunk 3]
```diff
@@ -16,19 +17,38 @@
...
+
     public OptionMenuPage clickCreatedAssetOptionBtn() {
         createdAssetOptionBtn.click();
         return new OptionMenuPage();
     }
 
+    public String getField(final String field) {
+        return driver.findElement(By.xpath(String.format(HEADER_ATTRIBUTE_LOCATOR, field))).getText();
+    }
+
```
Comment (Raymundogs) [inline src/main/java/salesforce/ui/pages/assets/AssetDetailPage.java:60]: Each method should have its javadoc.

File: src/main/java/salesforce/api/config/Credentials.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ */
+public enum Credentials {
+    
+    USERNAME1("USERNAME1", "username"),
+    PASSWORD("PASSWORD", "password"),
+    CLIENT_ID("CLIENT_ID", "client_id"),
+    CLIENT_SECRET("CLIENT_SECRET", "client_secret"),
+    GRANT_TYPE("password", "grant_type"),
+    TOKEN_URL("TOKEN_URL", "TOKEN_URL"),
+    BASE_URL("BASE_URL", "BASE_URL");
```
Comment (edsonarios) [inline src/main/java/salesforce/api/config/Credentials.java:22]: I think that "1" in "USERNAME1" is a typo, or is that right?
and the password is in lower case?

File: src/test/java/salesforce/assets/CreateAssetTest.java (hunk 1 above)
Comment (JRojas999) [reply src/test/java/salesforce/assets/CreateAssetTest.java]: You were right. I already changed it.

File: src/main/java/salesforce/ui/pages/assets/NewAssetPage.java (hunk 2 above)
Comment (JRojas999) [reply src/main/java/salesforce/ui/pages/assets/NewAssetPage.java:99]: You were right. I already added them.

File: src/main/java/salesforce/ui/pages/assets/AssetDetailPage.java (hunk 3 above)
Comment (JRojas999) [reply src/main/java/salesforce/ui/pages/assets/AssetDetailPage.java:60]: You were right. I already added them.

File: src/test/java/salesforce/assets/CreateAssetTest.java [hunk 5]
```diff
@@ -8,55 +8,90 @@
...
-        createAssetPage.clickRoleOption();
-        createAssetPage.clickRoleFirstOption();
-        createAssetPage.clickContactOption();
-        createAssetPage.clickContactFirstOption();
-        createAssetPage.clickProductOption();
-        createAssetPage.clickProductFirstOption();
-        createdAssetPage = createAssetPage.clickSaveBtn();
-        Assert.assertEquals(createdAssetPage.getCreatedAssetTitleText(), "Name Asset 1",
+        NewAssetPage createAssetPage = assetPage.clickCreateAssetBtn();
+        createAssetPage.setField("Asset Name", "Name Asset 1")
```
Comment (edsonarios) [inline src/test/java/salesforce/assets/CreateAssetTest.java:47]: Maybe the data you send could be parameters, so as not to repeat the same text

File: src/test/java/salesforce/assets/CreateAssetTest.java [hunk 6]
```diff
@@ -8,55 +8,90 @@
...
+
     @Test
-    public void testCreateAssetWithNecesaryAttributes() {
-        CreateAssetPage createAssetPage = assetPage.clickCreateAssetBtn();
-        createAssetPage.setUserName("Name Asset 1");
-        createAssetPage.clickRoleOption();
-        createAssetPage.clickRoleFirstOption();
-        createdAssetPage = createAssetPage.clickSaveBtn();
-        Assert.assertEquals(createdAssetPage.getCreatedAssetTitleText(), "Name Asset 1",
+    public void testCreateAssetWithNecessaryAttributes() {
```
Comment (edsonarios) [inline src/test/java/salesforce/assets/CreateAssetTest.java:27]: Maybe the data you send could be parameters, so as not to repeat the same text

File: src/main/java/salesforce/api/config/Credentials.java (hunk 4 above)
Comment (JRojas999) [reply src/main/java/salesforce/api/config/Credentials.java:22]: That typo is necessary because system environment variables.

**Pull request merged**